docs/entities/index.md

```diff
@@ -0,0 +1,449 @@
+---
+sidebar_position: 1
+---
+# Entities
+
+Entities are in-world objects that can interact with the world in a variety of ways. Common example include mobs, projectiles, rideable objects, and even players. Each entity consists of multiple systems that may not seem understandable at first glance. This section will break down some of the key components related to constructing an entity and making it behave as the modder intends.
+
+## Terminology
+
+A simple entity is made up of three parts:
+
+- The [`Entity`][entity] subclass, which holds most of our entity's logic
+- The [`EntityType`][type], which is [registered][registration] and holds some common properties, and
+- The [`EntityRenderer`][renderer], which is responsible for displaying the entity in-game
+
+More complex entities may require more parts. For example, many of the more complex `EntityRenderer`s use an underlying `EntityModel` instance. Or, a naturally spawning entity will need some sort of [spawn mechanism][spawning].
+
+## `EntityType`
+
+The relationship between `EntityType`s and `Entity`s is similar to that of [`Item`s][item] and [`ItemStack`s][itemstack]. Like `Item`s, `EntityType`s are singletons that are registered to their corresponding registry (the entity type registry) and hold some values common to all entities of that type, while `Entity`s, like `ItemStack`s, are "instances" of that singleton type that hold data specific to that one entity instance. However, the key difference here is that most of the behavior is not defined in the singleton `EntityType`, but rather in the instantiated `Entity` class itself.
+
+Let's create our `EntityType` registry and register an `EntityType` for it, assuming we have a class `MyEntity` that extends `Entity` (see [below][entity] for more information). All methods on `EntityType.Builder`, except for the `#build` call at the end, are optional.
```

Comment:
A lot of the entity pages reads like a tutorial, rather than documentation. I am not sure on the guidelines for this, so if this is invalid critique, please ignore. Especially since I see similar tutorial-esque language used in other parts of documentation already as well.

For example, the line in question reads "Let's create our `EntityType` registry....".
Documentation should be more of "You can to register an `EntityType` similarly to `Item`s using `DeferredRegister.Entities`....." Or something similar.
Note the difference in language of "lets do this" or "we will" vs "You can" or "... is done by doing ..."

Reply:
While I would prefer such language as well, it is not a hard requirement for a contribution. If the language is particularly misleading, then we can adjust it or add such language to the contributing section. But for right now, it's fine.

Reply:
I generally try using more tutorial-like language in the documentation because it makes it easier for new users on one hand, and on the other hand because it then serves as a kind of to-do list for adding new stuff (since adding new stuff is notoriously split across multiple places in Minecraft/NeoForge).